File: 2023/02/01.ts
//const doc = require('fs').readFileSync('example.txt').toString().split('\n');
const doc = require('fs').readFileSync('input.txt').toString().split('\n');

// 12 red cubes, 13 green cubes, and 14 blue cubes
function checkPossibility(num, color) {
    switch (color) {
        case 'red':
            if (num > 12)
                return false;
            break;
        case 'green':
            if (num > 13)
                return false;
            break;
        case 'blue':
            if (num > 14)
                return false;
            break;
    }
    return true;
}

function calibrate() {
    let res = 0;
    doc.forEach((line, i) => {
        let set = i + 1
        line.split(': ')[1].split('; ').forEach(sets => {
            sets.split(', ').forEach(cubes => {
                const possible = checkPossibility(parseInt(cubes.split(' ')[0]), cubes.split(' ')[1])
                if (!possible) set = 0
            })
        })
        res += set
    });
    return res
}

console.log('Answer: ' + calibrate())

File: 2024/day07/01.py
calib = []


def checkOperator(result, numbers) -> bool:
    if sum(numbers) == result:
        return True

    res = 1
    for val in numbers:
        res = val * res

    if res == result:
        return True

    return False


# with open("input.txt", "r") as file:
with open("example.txt", "r") as file:
    for line in file:
        result = int(line.split(": ")[0])
        numbers = list(map(int, line.split(": ")[1].split(" ")))
        if checkOperator(result, numbers):
            calib.append(result)


print(sum(calib))
